src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -1076,6 +1142,35 @@ def _get_latest_revision(bucket_paths):
   return None
 
 
+def _maybe_get_http_build_url(revision) -> Optional[str]:
+  """Tries to get the HTTP build URL if it exists.
+
+  Args:
+      revision: the revision we're currently handling.
+
+  Returns:
+      the build URL.
+  """
+  http_build_url_pattern = environment.get_value('RELEASE_BUILD_URL_PATTERN')
+  if not http_build_url_pattern:
+    return None
+  dir_name = os.path.dirname(http_build_url_pattern)
+  basename = os.path.basename(http_build_url_pattern)
```

Comment:
If you want to use the path functions, I think you really should use posixpath, because on windows, I think this breaks.